### ISSUE TITLE
why not divide by batch size ?

### ISSUE THREAD

**Issue by Johnson-yue** [pytorch/examples#652]
https://github.com/pytorch/examples/blob/4e00723456160d910092aae567a0b8daf66c49ec/vae/main.py#L82

I think finally loss should be **(BCE+KLD) / batch_size** , is right?

**Comment by nbqu** [comment]
https://jaan.io/what-is-variational-autoencoder-vae-tutorial/

maybe the size of the batch doesn't related to it.

**Comment by rjkilpatrick** [comment]
`(BCE+KLD) / args.batch_size` would result in diving the loss function by a constant, which is a scaling operation which does not affect the loss function solutions.
See the original paper <https://arxiv.org/pdf/1312.6114.pdf> (equation 10) for a discussion about the specifics of the loss function
